### PR TITLE
198 add val data to training revised

### LINKED ISSUE #159: Clean up/finalize baseline LSTM
This issue captures several tasks that are needed to clean up/finalize the baseline LSTM:

- [x] Add `tmmn` to list of meteorological drivers #153
- [x] Check date formatting for gridmet data
- [ ] Incorporate gridmet data in targets pipeline #151 
- [x] Add additional segment/catchment attributes at NHDPlusv2-scale #51
- [x] `inputs_and_outputs_med_obs` in `2a_model.R` has redundant COMID columns
- [x] Simplify pipeline by omitting targets for _both_ well-observed and medium-observed sites #148 
- [x] Decide whether provisional data will be included in the observations used for model training #135 
- [x] Val times are currently defined in two places, and the values differ between those locations #160 
- [ ] Flesh out README to describe the modeling inputs and steps, including how to modify and run the model #146

### LINKED ISSUE #166: Decide how to handle test sites in analysis and plotting workflows
We want to create workflows that will transfer easily to the test sites.

### LINKED ISSUE #198: Incorporate validation data into train set, evaluate on test
We need to incorporate what has been the validation data into the train set and evaluate on the test set. Once this is done, we will not be making any more modifications to the models.

### LINKED ISSUE #204: Training start date on calendar year
I noticed today that the training start date is at the beginning of the _calendar_ year, not the _water_ year. Seems like it should be the _water year_

https://github.com/USGS-R/drb-do-ml/blob/43b4277f0fe3ae4d3f61dc8bb48ed5a78b36f456/_targets.R#L91

### PR DESCRIPTION
**Pull request by jsadler2**
** This PR supercedes #202. Updated version of that after our discussion on our 3/1 meeting. **

I moved the old validation data into the training set and I moved the old "test" set into the validation set.

This also starts the training on the water year (#204) and excludes any provisional data (#135).

 ⚠️ This PR will constitute the final results for the paper since we would no longer have a test set.

closes #198
closes #204
closes #159 
closes #166

### CODE REVIEW

File: 2a_model/src/models/config_base.yml [hunk 1]
```diff
@@ -10,47 +10,41 @@ recurrent_dropout: 0.2
...
-train_start_date: '1980-01-01'
-train_end_date: '2014-10-01'
-val_start_date: '2014-10-01'
-val_end_date: '2015-10-01'
-test_start_date: '2015-10-01'
-test_end_date: '2022-10-01'
+  - "01475530"
+  - "01475548"
+train_start_date: '1980-10-01'
+train_end_date: '2015-10-01'
```
Comment (lekoenig) [inline 2a_model/src/models/config_base.yml:16]: It looks like from `p2a_site_splits` that records on `2015-10-01` are treated as validation data, is that right?

**Review by lekoenig — APPROVED** [review]
Thanks, @jsadler2! I have a couple pretty minor questions just to confirm some of the assignments. I feel comfortable merging this PR whenever you are ready.

**Review by galengorski — APPROVED** [review]

File: 2a_model/src/models/Snakefile_base.smk [hunk 2]
```diff
@@ -124,8 +120,7 @@ rule make_predictions:
 def filter_predictions(all_preds_file, partition, out_file):
         df_preds = pd.read_feather(all_preds_file)
         all_sites = df_preds.site_id.unique()
-        trn_sites = all_sites[(~np.isin(all_sites, config["validation_sites"])) &
-                              (~np.isin(all_sites, config["test_sites"]))]
+        trn_sites = all_sites[~np.isin(all_sites, config["validation_sites"])]
```
[the remark this answers is not in the document]
Comment (lekoenig) [reply 2a_model/src/models/Snakefile_base.smk:123]: @jsadler2 confirmed that the `~` refers to the inverse so what this line is doing is subsetting `all_sites` for the sites that are not within the `validation_sites` vector from the config file.

File: 2a_model/src/models/config_base.yml (hunk 1 above)
Comment (jsadler2) [reply 2a_model/src/models/config_base.yml:16]: Yes.
